pyrefly/lib/test/lsp/code_actions.rs

```diff
@@ -45,6 +49,128 @@ fn get_test_report(state: &State, handle: &Handle, position: TextSize) -> String
     report
 }
 
+fn get_test_report_ignore_code_action(
+    state: &State,
+    handle: &Handle,
+    position: TextSize,
+) -> String {
+    let mut report = "Code Actions Results:\n".to_owned();
+    let transaction = state.transaction();
+    for (title, info, range, patch) in transaction
+        .local_quickfix_code_actions(handle, TextRange::new(position, position))
+        .into_iter()
+        .flatten()
+        // NOTE: We ONLY patch buffer with ignore comment code action so
+        // we can test it separately
+        .filter(|(title, _, _, _)| title.contains("ignore"))
+    {
+        let (before, after) = apply_patch(&info, range, patch);
+        report.push_str("# Title: ");
+        report.push_str(&title);
+        report.push('\n');
+        report.push_str("\n## Before:\n");
+        report.push_str(&before);
+        report.push_str("\n## After:\n");
+        report.push_str(&after);
+        report.push('\n');
+    }
+    report
+}
+
+#[test]
+fn add_ignore_comment() {
```

Comment:
could this test instead be multiple smaller tests that each test something different? I don't know if you need multiple files either for any of these

- add_ignore_comment_basic
- add_ignore_comment_multiline
- add_ignore_indentation (what happens when the indentation changes?)
- add_to_existing_ignore_basic
- add_to_existing_ignore_multiline
- add_to_existing_ignore_indentation (what happens here when indentation is different?)